core/block/source/service.go

```diff
@@ -133,6 +136,19 @@ func (s *service) newSource(ctx context.Context, space Space, id string, buildOp
 			return NewBundledRelation(id), nil
 		case smartblock.SmartBlockTypeIdentity:
 			return NewIdentity(s.identityService, id), nil
+		case smartblock.SmartBlockTypeParticipant:
+			participantState := state.NewDoc(id, nil).(*state.State)
+			participantState.SetObjectTypeKey(bundle.TypeKeyParticipant)
```

Comment:
why not in editor/participant.go? just interesting

Reply:
because of deriving type id, I'll write a comment